float64 encoder0
float64 encoder1
float64 encoder2
float64 encoder3
